pact/Pact/Core/Gas.hs

```diff
@@ -246,15 +231,44 @@ constantGasModel unitPrice gl
   }
 
 freeGasModel :: GasModel b
-freeGasModel = constantGasModel mempty (MilliGasLimit mempty)
+freeGasModel = constantGasModel mempty (MilliGasLimit (MilliGas maxBound)) -- TODO: some tests seem to charge gas even with freeGasModel.
```

Comment:
Is this because we're charging gas directly?

Maybe GasM needs to account for the "free gas model"...

Maybe the free gas model should default to a massive gas limit instead?

Reply:
That's it, yes - `GasM` charges gas directly, rather than looking up the values from any `GasModel`.

A quick and dirty solution would be keeping the change currently in the PR - `freeGasModel` just has an infinite limit. But that's a hack.

Probably a better solution is for `GasM` to carry a `GasModel` in its `GasMEvn`, and for the serialization factors to be gas args, like all other gassed operations. WDYT @jmcardon ?

Reply:
I've derived `Bounded` for `MilliGasLimit` in  https://github.com/kadena-io/pact-5/blob/37633735c86f7eedaf55800f4670570eb2c88603/pact/Pact/Core/Gas.hs#L68 

In my opinion, we should make `freeGasModel :: GasModel b` consistent with `tableGasModel :: MilliGasLimit -> GasModel CoreBuiltin`, i.e. taking `MilliGasLimit`.

Reply:
I added a table of serialization gas costs to `GasModel`, and added `GasModel` to `GasMEnv`. Now we can specify that serialization is free within the `freeGasModel`, and those costs get propagated through to the code that runs serialization.